### PR TITLE
feat: add code preview snippets

### PR DESCRIPTION
**Pull request by tinaszheng**
image

# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
The focus of this PR is to update the `ui-demo` package by adding the `dedent` package, refactoring context types, and improving authentication configuration handling.

### Detailed summary
- Added `dedent` package to `package.json`
- Refactored `ConfigContext` type to `ConfigContextType`
- Updated authentication configuration handling in components
- Utilized `useConfig` hook in `Authentication` and `CodePreview` components

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: examples/ui-demo/src/components/preview/CodePreview.tsx [hunk 1]
```diff
@@ -50,9 +52,41 @@ function CodeBlock({ title, code }: { title: string; code: string }) {
   );
 }
 
-const styleCode = `import { withAccountKitUi } from "@alchemy/aa-alchemy/tailwind";
+
+function getTailwindCode(config: Config) {
+  const { ui } = config
+  // TODO: separate primaryColor for light and dark mode
+  return `import { withAccountKitUi, createColorSet } from "@alchemy/aa-alchemy/tailwind";
```
Comment (moldy530) [inline examples/ui-demo/src/components/preview/CodePreview.tsx:59]: nit: oh useful util we use in a few other places is the `dedent` package which makes it much easier to get these things formatted in code nicely. basically lets you do:
```
return dedent`
import ...

....
`;
```

File: examples/ui-demo/src/components/preview/CodePreview.tsx [hunk 2]
```diff
@@ -50,9 +52,41 @@ function CodeBlock({ title, code }: { title: string; code: string }) {
...
+
+  sections.push([{ type: "passkey" }])
+
+  return `const uiConfig = {
+  auth: {
+    sections: ${JSON.stringify(sections)},
+    addPasskeyOnSignup: ${config.auth.addPasskey},
+  },
+};
+  `
```
Comment (moldy530) [inline examples/ui-demo/src/components/preview/CodePreview.tsx:91]: I think this should include the full code snippet including the `createConfig` call so maybe something like:

```
export const config = createConfig({
 // the default config we recommend in the examples for that method (think rpcUrl: /api/rpc and chain: sepolia
});
```

File: examples/ui-demo/src/components/preview/CodePreview.tsx [hunk 3]
```diff
@@ -50,9 +52,41 @@ function CodeBlock({ title, code }: { title: string; code: string }) {
...
+function getTailwindCode(config: Config) {
+  const { ui } = config
+  // TODO: separate primaryColor for light and dark mode
+  return `import { withAccountKitUi, createColorSet } from "@alchemy/aa-alchemy/tailwind";
 import type { Config } from "tailwindcss";
 
-const config: Config = <your tailwind config>;
-export default withAccountKitUi(config);
-`;
+export default withAccountKitUi(<your tailwind config>, {
```
Comment (moldy530) [inline examples/ui-demo/src/components/preview/CodePreview.tsx:62]: something I realized we might need in the alpha getting started doc as we well was the a comment linking to `tailwind` getting started docs, we should add that here as well

**Review by moldy530 — APPROVED** [review]
LGTM, just a few nits. dedent might be nice for the longer code snippets, but not super high prio (also I haven't checked the bundle size if it since we only use it as a dev dep in most places or it's used in CLIs where bundle doesn't matter that much)

File: examples/ui-demo/src/components/preview/CodePreview.tsx (hunk 1 above)
Comment (tinaszheng) [reply examples/ui-demo/src/components/preview/CodePreview.tsx]: oh good idea, added!

**Pull request merged**